staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply.go

```diff
@@ -65,12 +55,12 @@ func (p *applyPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (
 	managedFields := accessor.GetManagedFields()
 	accessor.SetManagedFields(nil)
 
-	currentTyped, err := pType.FromUnstructured(current)
+	currentTyped, err := p.converter.ObjectToTyped(currentObject)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create typed current object: %v", err)
 	}
 
-	newTyped, err := pType.FromYAML(typed.YAMLObject(p.patchBytes))
```

Comment:
Yeah, you can't do that, but I suspect it'd be easy to improve?

Reply:
don't know how you were able to add the comment here as it's already outdated. But the new version is using the newObject I added to the call.
Would be good to know if that even works and if/why it's not good.